src/views/List.jsx

```diff
@@ -1,23 +1,41 @@
+import { useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchTerm, setSearchTerm] = useState('');
+
+	/* TO DO: Make separate resuable input component with a filter feature*/
+
+	const filterList = data.filter((item) => {
+		if (searchTerm === '') {
+			return item;
+		} else if (item.name.toLowerCase().includes(searchTerm.toLowerCase())) {
+			return item;
+		}
+	});
+
+	const renderList = filterList.map((item) => (
+		<ListItem name={item.name} key={item.id} />
+	));
+
+	const handleSubmit = (e) => {
+		e.preventDefault();
+		setSearchTerm('');
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
-			<ul>
-				{
-					/**
-					 * TODO: write some JavaScript that renders the `data` array
-					 * using the `ListItem` component that's imported at the top
-					 * of this file.
-					 */
-					data.map((item) => (
-						<ListItem name={item.name} key={item.id} />
-					))
-				}
-			</ul>
+			<form onSubmit={handleSubmit}>
+				<label htmlFor="search-filter">Item name:</label>
+				<input
+					type="text"
+					id="search-filter"
+					value={searchTerm}
+					onChange={(e) => setSearchTerm(e.target.value)}
+				/>
+				<button type="submit"> X </button>
```

Comment:
🔍 Great job getting this button to work!

As people who are accustomed to common patterns on the internet, we can see an "X" button and guess that it means something like "close" or "clear", but not everyone can parse this kind of ambiguity. Rather than leaving the user to guess, I think we could use the word "Clear" in our design.

Reply:
That change makes sense to me. An X could mean all kinds of things, "close, clear, cancel." Will fix. Thanks!